README.md

```diff
@@ -70,18 +70,85 @@ DeclareSchema::Migration::Migrator.before_generating_migration do
+The default text limit can be set using the `Generators::DeclareSchema::Migration::Migrator.default_text_limit=` method.
+Note that a `nil` default means that there is no default-- so every declaration must be explicit.
+This will `raise` a `limit: must be provided for field :text...` error when the default value is `nil` and there is no explicit
+declaration.
 
-### Global Configuration
+For example, adding the following to your `config/initializers` directory will
+set the default `text limit` value to `0xffff_ffff`:
+ 
+**declare_schema.rb**
+```ruby
+# frozen_string_literal: true
+
+Generators::DeclareSchema::Migration::Migrator.default_text_limit = 0xffff_ffff
+```
+
+#### String Limit
+The default string limit can be set using the `Generators::DeclareSchema::Migration::Migrator.default_string_limit=` method.
+Note that a `nil` default means that there is no default-- so every declaration must be explicit.
+This will `raise` a `limit: must be provided for field :string...` error when the default value is `nil` and there is no explicit
+declaration.
+
+For example, adding the following to your `config/initializers` directory will
+set the default `string limit` value to `nil`:
+ 
+**declare_schema.rb**
+```ruby
+# frozen_string_literal: true
+
+Generators::DeclareSchema::Migration::Migrator.default_string_limit = nil
+```
+
+#### Null
+The default null value can be set using the `Generators::DeclareSchema::Migration::Migrator.default_null=` method.
+Note that a `nil` default means that there is no default-- so every declaration must be explicit.
+This will `raise` a `null: must be provided for field...` error when the default value is `nil` and there is no explicit
+declaration.
+
+For example, adding the following to your `config/initializers` directory will
+set the default `null` value to `false`:
+ 
+**declare_schema.rb**
+```ruby
+# frozen_string_literal: true
+
+Generators::DeclareSchema::Migration::Migrator.default_null = false
+```
+
+#### Generate Foreign Keys
+The default value for generate foreign keys can be set using the `Generators::DeclareSchema::Migration::Migrator.default_generate_foreign_keys=` method.
+This value defaults to `true` and can only be set at the global level.
+
+For example, adding the following to your `config/initializers` directory will
+generate foreign keys:
+ 
+**declare_schema.rb**
+```ruby
+# frozen_string_literal: true
+
+Generators::DeclareSchema::Migration::Migrator.default_generate_foreign_keys = false
+```
+
+#### Generate Indexing
+The default value for generate indexing can be set using the `Generators::DeclareSchema::Migration::Migrator.default_generate_indexing=` method.
+This value defaults to `true` and can only be set at the global level.
+
+For example, adding the following to your `config/initializers` directory will
+generate indexing:
+ 
+**declare_schema.rb**
+```ruby
+# frozen_string_literal: true
+
+Generators::DeclareSchema::Migration::Migrator.default_generate_foreign_keys = false
```

Comment:
I think this is a copy/paste error? But also, like above, I'd make the example show how to configure different from the default. I think the default here is `true` for generating indexes.